gpkit/nomials/substitution.py

```diff
@@ -125,11 +125,20 @@ def substitution(nomial, substitutions, val=None):
             if len(varlocs_[var]) == 0:
                 del varlocs_[var]
             if isinstance(sub, Numbers):
+                if isinstance(sub, Quantity):
```

Comment:
I think `isinstance` is slower than `hasattr` (and that hasattr is more pythonic).

Should we check for units attr here instead?